deploy/supervisor/values.yaml

```diff
@@ -98,23 +99,31 @@ no_proxy: "$(KUBERNETES_SERVICE_HOST),169.254.169.254,127.0.0.1,localhost,.svc,.
 #!     network: tcp
 #!     address: :8443
 #!   http:
-#!     network: tcp
-#!     address: :8080
-#!
-#! These defaults mean: bind to all interfaces using TCP.  Use port 8443 for https and 8080 for http.
-#! The defaults will change over time.  Users should explicitly set this value if they wish to avoid
-#! any changes on upgrade.
-#!
-#! A future version of the Supervisor app may include a breaking change to adjust the default
-#! behavior of the http listener to only listen on 127.0.0.1 (or perhaps even to be disabled).
+#!     network: disabled
 #!
-#! Binding the http listener to addresses other than 127.0.0.1 or ::1 is deprecated.
+#! These defaults mean: For HTTPS listening, bind to all interfaces using TCP on port 8443.
+#! Disable HTTP listening by default.
 #!
-#! Unix domain sockets are recommended for integrations with service meshes.  Ingresses that terminate
-#! TLS connections at the edge should re-encrypt the data and route traffic to the https listener.
+#! The HTTP listener can only be bound to loopback interfaces. This allows the listener to accept
+#! traffic from within the pod, e.g. from a service mesh sidecar. The HTTP listener should not be
+#! used to accept traffic from outside the pod, since that would mean that the network traffic could be
+#! transmitted unencrypted. The HTTPS listener should be used instead to accept traffic from outside the pod.
+#! Ingresses and load balancers that terminate TLS connections should re-encrypt the data and route traffic
+#! to the HTTPS listener. Unix domain sockets may also be used for integrations with service meshes.
 #!
-#! Changing the port numbers used must be accompanied with matching changes to the service and deployment
-#! manifests.  Changes to the https listener must be coordinated with the deployment health checks.
+#! Changing the HTTPS port number must be accompanied by matching changes to the service and deployment
+#! manifests. Changes to the HTTPS listener must be coordinated with the deployment health checks.
 #!
 #! Optional.
 endpoints:
+
+#! Optionally override the validation on the endpoints.http value which checks that only loopback interfaces are used.
+#! When deprecated_insecure_accept_external_unencrypted_http_requests is true, the HTTP listener is allowed to bind to any
+#! interface, including interfaces that are listening for traffic from outside the pod. This value is being introduced
+#! to ease the transition to the new loopback interface validation for the HTTP port for any users who need more time
+#! to change their ingress strategy to avoid using plain HTTP into the Supervisor pods.
+#! This value is being introduced and immediately deprecated. It will be removed in some future release, at which time
+#! traffic from outside the pod will need to be sent to the HTTPS listener instead, with no simple workaround available.
+#! Allowed values are true (boolean), "true" (string), false (boolean), and "false" (string). The default is false.
```

Comment:
Why do we need to support non-boolean inputs here?

Reply:
We don't need to, but the idea was to make it easier for ytt users because they can either use `--data-value deprecated_insecure_accept_external_unencrypted_http_requests=true` (a string) or `--data-value-yaml deprecated_insecure_accept_external_unencrypted_http_requests=true` (a boolean).